internal/topology/variables/patch_validation.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package variables
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"text/template"
+
+	"github.com/dolmen-go/jsonptr"
```

Comment:
This repository seems unmaintained

/hold

Reply:
It's tricky and it depends how much we want to actually validate per the RFC. There's an open api library https://github.com/go-openapi/jsonpointer but it doesn't do the same level of validation. It's probably good enough in this case.